nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java

```diff
@@ -289,37 +306,24 @@ public String toString() {
 
 
     private void detectClassAndMethod(final LogRecord wrappedRecord) {
-        final StackTraceElement[] stack = Thread.currentThread().getStackTrace();
-        boolean ignoredClass = false;
-        for (final StackTraceElement element : stack) {
-            final String className = element.getClassName();
-            if (!ignoredClass) {
-                ignoredClass = isIgnoredStackTraceElement(className);
-                continue;
-            }
-            if (!isIgnoredStackTraceElement(className)) {
-                wrappedRecord.setSourceClassName(className);
-                wrappedRecord.setSourceMethodName(element.getMethodName());
-                return;
-            }
-        }
+        final Optional<StackWalker.StackFrame> stackFrame = STACK_WALKER.walk(stackFrames ->
+            stackFrames.dropWhile(frame -> !isIgnoredStackFrame(frame.getClassName()))
+                .filter(frame -> !isIgnoredStackFrame(frame.getClassName()))
+                .findFirst());
```

Comment:
I'd drop local variable `stackFrame`, continue with `.ifPresent`, and remove import for `Optional`.